Source/WebCore/page/Navigation.cpp

```diff
@@ -525,4 +531,24 @@ bool Navigation::dispatchDownloadNavigateEvent(const URL&, const String& downloa
     return false;
 }
 
+JSC::JSObject* convertDictionaryToJS(JSC::JSGlobalObject& lexicalGlobalObject, JSDOMGlobalObject& globalObject, const Navigation::Result& dictionary)
```

Comment:
There are 0 other places where we needed custom convertDictionaryToJS functions in WebKit.  If a change is needed, it should be an IDL attribute.  Why doesn't the generated function work?

Reply:
That is true. Note though that returning a promise pair is new, not well specced and likely needs some creative solution.
I thought of a few and I am not sure if they are viable:
- hardcode callPromisePairFunction to deal with returning a JS object consisting of two JS Promises (this was the initial approach). Problem is that it would be hacky to just hardcode the committed/finished names., whereas for callPromiseFunction it is just an anonymous name
- weaken the assertion in DOMPromise::promise to ignore isSuspended if the JS promise is already settled (could be checked in JSConverter<IDLPromise<T>> too). I don't know if that is acceptable outside of Navigation API though.
- what I did in my new PR. In general we could add more keywords for promise pair to the idl and react on them in the code generator, but I don't see a particular clean way. 

Of course, feel free to suggest yet another solution :)

Reply:
Why can’t you just use an attribute on the type instead of a hardcoded type name? That’s all @achristensen07 is asking for, and I don’t see why it’s not practical.

Reply:
Ok, I didn't read Alex comment carefully yesterday, at least not the " If a change is needed, it should be an IDL attribute." part. I'll experiment a bit with that.